pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -422,7 +422,7 @@ func (r *ReconcileClusterDeployment) reconcile(request reconcile.Request, cd *hi
 		return reconcile.Result{Requeue: true}, nil
 	}
 
-	if cd.Status.InstallerImage == nil {
+	if cd.Status.Installed && (cd.Status.InstallerImage == nil || cd.Status.CLIImage == nil) {
```

Comment:
I don't follow here. Why do we only want to resolve the installer image when the cluster has already been installer?

Reply:
Yeah it was quite wrong when I first pushed, should be !Installed.